### ISSUE TITLE
Multi-languages setup.

### ISSUE THREAD

**Issue by JorgeAnzola** [Rias500/craft-scout#25]
As you might remember I'm using the plugin for a multi-languages setup, add first I made a [PR](https://github.com/Rias500/craft-scout/pull/21) to change the id and prepend the site ID to the objectId. But that's not enough to have a localised search. The [docs recommend to have an index per languages](url).  

**I'm really not saying this commit in my fork is the solution**, it works for me, but might be a hint or an idea for you?

Please check [this commit](https://github.com/JorgeAnzola/craft-scout/commit/a53179aca5c0c240bbc205f0a5a4f6e9c53f8cc0#diff-0ac269d97c48a3a770661c2aecf33a2cR106).

Then in the config I do this:

```
[
            'indexName' => 'news__en_us',
            'elementType' => Entry::class,
            'criteria' => [
                'section' => 'news',
                'site' => 'en_us'
            ],
            'transformer' => new SearchIndexEntryTransformer(),
        ],
        [
            'indexName' => 'news__nl_nl',
            'elementType' => Entry::class,
            'criteria' => [
                'section' => 'news',
                'site' => 'nl_nl'
            ],
            'transformer' => new SearchIndexEntryTransformer(),
        ],
```

**Comment by JorgeAnzola** [comment]
Nevermind, with the criteria is enough, apparently.

**Comment by Rias500** [comment]
This should work with just the criteria, as Scout checks if the query returns anything before sending it to Algolia.

I have a site running with pretty much the same config and 2 separate indexes for multilanguage

**Comment by JorgeAnzola** [comment]
Okok, I imaged. I actually deleted my comment saying "Nevermind it works with the Criteria. Cause I was not sure. I'll give it a try.

**Comment by JorgeAnzola** [comment]
@Rias500 I cannot get this to work:

Do you see something wrong?

```
[
            'indexName' => 'shows__en_us',
            'elementType' => Entry::class,
            'criteria' => [
                'section' => 'shows',
                'site' => 'en_us',
            ],
            'transformer' => new SearchIndexEntryTransformer(),
        ],
        [
            'indexName' => 'shows__nl_nl',
            'elementType' => Entry::class,
            'criteria' => [
                'section' => 'shows',
                'site' => 'nl_nl',
            ],
            'transformer' => new SearchIndexEntryTransformer(),
        ],
```

**Comment by Rias500** [comment]
Are you sure you're using the site `handle` and not the language by any chance?

**Comment by JorgeAnzola** [comment]
Yes, I'm sure. As a test, I changed `en_us` to `en_us2`, getting this error:

```
Invalid site handle: en_us2
```

It will only save the secondary site version of the entry. (I'm using 0.4.1, without the ID change, as maybe there's a bug on the change I requested)

**Comment by Rias500** [comment]
Okay, I'll need to take a better look at it then. I'll try to find a solution this weekend

**Comment by JorgeAnzola** [comment]
Thanks! I'll also try to find something.

**Comment by JorgeAnzola** [comment]
And about the site id in the objectId: This changes fixes my problem with the multi site, but then during the deindexation causes issues like: deindexing from just one index.

**Comment by nlussier-globalia** [comment]
Same problem here : English element is indexed with only the objectId in my French index and vice-versa.

**Comment by Rias500** [comment]
This should finally be fixed in 1.0.0, sorry for the long wait!
